File: app/src/main/java/org/bspb/smartbirds/pro/ui/fragment/OsmMapFragment.java
package org.bspb.smartbirds.pro.ui.fragment;

import androidx.fragment.app.Fragment;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.EFragment;
import org.androidannotations.annotations.ViewById;
import org.bspb.smartbirds.pro.R;
import org.bspb.smartbirds.pro.events.EEventBus;
import org.bspb.smartbirds.pro.events.MapAttachedEvent;
import org.osmdroid.views.MapView;

/**
 * Created by dani on 14-11-4.
 */
@EFragment(R.layout.fragment_map_osm)
public class OsmMapFragment extends Fragment {

    @ViewById(R.id.mapview)
    MapView mapView;

    EEventBus eventBus = EEventBus.getInstance();

    public MapView getMapView() {
        return mapView;
    }

    @AfterViews
    public void init() {
        eventBus.post(new MapAttachedEvent());
    }
}

File: app/src/main/kotlin/org/bspb/smartbirds/pro/ui/fragment/OsmMapFragment.kt
package org.bspb.smartbirds.pro.ui.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import org.bspb.smartbirds.pro.R
import org.bspb.smartbirds.pro.events.EEventBus
import org.bspb.smartbirds.pro.events.MapAttachedEvent
import org.osmdroid.views.MapView

class OsmMapFragment : Fragment() {

    var mapView: MapView? = null
        private set
    private var eventBus: EEventBus = EEventBus.getInstance()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return super.onCreateView(inflater, container, savedInstanceState) ?: inflater.inflate(
            R.layout.fragment_map_osm,
            container,
            false
        )
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        mapView = view.findViewById(R.id.mapview)
        eventBus.post(MapAttachedEvent())
    }
}